app/models/rent.rb

```diff
@@ -0,0 +1,15 @@
+class Rent < ApplicationRecord
+  belongs_to :user
+  belongs_to :book
+
+  validates :user, presence: true
+  validates :book, presence: true
+  validates :from, presence: true
+  validates :to, presence: true
+
+  validate :validate_date_from_is_less_than_date_to
+
+  def validate_date_from_is_less_than_date_to
+    errors.add(:to, 'Date from must be less than date to') if from && to && to < from
```

Comment:
You can use an `if: {}` block in the validation definition to avoid checking the existence of `from` and `to` here. 
```ruby
validate :validate_date_from_is_less_than_date_to, if: -> { from.present? && to.present? }
```

This way you keep separated the logic of the validation (`to < from`) and the prerequisites to run that validation (existence of `from` and `to`).
You don't need to make this change, is just a suggestion but keep it in mind!